Resources/frontend/js/jquery.adyen-payment-selection.js

```diff
@@ -328,13 +322,77 @@
-            ) ?? {};
+        __isGiftCardType: function (paymentType) {
+            var me = this;
+            var paymentGroups = me.opts.adyenPaymentMethodsResponse['groups'] || [];
+            var filteredGiftCardGroup = paymentGroups.filter(function (group) {
+                return me.defaults.giftCardGroupName === group['name'];
+            });
+            var giftCardGroup = filteredGiftCardGroup[0] || [];
+            var giftCardGroupTypes = (giftCardGroup && giftCardGroup['types']) || [];
+            var filteredTypes = giftCardGroupTypes.filter(function(giftCardGroupType) {
+                return giftCardGroupType === paymentType;
+            });
 
-            return (giftCardGroup['types'] ?? []).includes(paymentType);
+            return filteredTypes.length > 0;
         },
+        /**
+         * paymentType contains the id for a stored payment
+         * @param {string} paymentMethodId
+         * @return {boolean}
+         * @private
+         */
+        __isStoredPaymentMethod: function (paymentMethodId) {
+            return !!this.getStoredPaymentMethodById(paymentMethodId);
+        },
+        __canHandlePayment: function (paymentMethod) {
+            if (this.__isStoredPaymentMethod(paymentMethod.id || '')) {
+                return true;
+            }
+
+            return "undefined" !== typeof paymentMethod.details;
+        },
+        /**
+         * Modify AdyenPaymentMethod with additional data for the web-component library
+         * @param paymentMethod Adyen response: Payment Method response
+         * @return  {{cardType: string, paymentMethodData: object}}
+         * @private
+         */
+        __buildCheckoutComponentData: function (paymentMethod) {
+            var defaultData = {
+                cardType: paymentMethod.type,
+                paymentMethodData: paymentMethod
+            };
+
+            if (this.__isStoredPaymentMethod(paymentMethod.id || '')) {
+                return $.extend(true, {}, defaultData, {
+                    paymentMethodData: {
+                        storedPaymentMethodId: paymentMethod.id
+                    }
+                });
+            }
+
+            if (this.__isGiftCardType(paymentMethod.type)) {
+                var pinRequiredDetail = this.__retrievePaymentMethodDetailByKey(
+                    paymentMethod.type,
+                    'encryptedSecurityCode'
+                );
+
+                return $.extend(true, {}, defaultData, {
+                    cardType: 'giftcard',
+                    paymentMethodData: {
+                        type: paymentMethod.type,
+                        pinRequired: false === pinRequiredDetail.optional || false
+                    }
+                });
+            }
+
+            return $.extend(true, {}, defaultData, {
+                paymentMethodData: {
+                    enableStoreDetails: paymentMethod.supportsRecurring || false
```

Comment:
@bortefi could you modify this so that `enableStoreDetails` is always true for scheme/cards? We always want to show the Save... checkbox for cards if the user is logged in regardless of the `supportsRecurring` flag in the methods response